src/net/sf/mpxj/primavera/PrimaveraPMFileReader.java

```diff
@@ -639,42 +639,45 @@ private void updateDates(Task parentTask)
          {
             updateDates(task);
 
-            if (actualStartDate == null || DateHelper.compare(actualStartDate, task.getActualStart()) > 0)
+            // the child tasks can have null dates (e.g. for nested wbs elements with no task children) so we
+            // still must protect against some children having null dates
+
+            if (actualStartDate == null || (task.getActualStart() != null && DateHelper.compare(actualStartDate, task.getActualStart()) > 0))
             {
                actualStartDate = task.getActualStart();
             }
 
-            if (actualFinishDate == null || DateHelper.compare(actualFinishDate, task.getActualFinish()) < 0)
+            if (actualFinishDate == null || (task.getActualFinish() != null && DateHelper.compare(actualFinishDate, task.getActualFinish()) < 0))
```

Comment:
I like the idea of adding a null-safe `DateHelper.min/max`. I'm sure there are other places in the code we could use it. I was also wondering about factoring out the `updateDates` methods from the `PrimaveraPMFileReader` and `PrimaveraReader` classes into a single implementation in a helper class. I think we can just use the code from `PrimaveraReader`.